ovs/openflow.go

```diff
@@ -267,13 +267,44 @@ func (o *OpenFlowService) DumpTables(bridge string) ([]*Table, error) {
 	return tables, err
 }
 
+func addArgsForDumpFlows(args []string, ofctlFlags []string, flow *MatchFlow) ([]string, error) {
+	args = append(args, ofctlFlags...)
+	if flow != nil {
+		fb, err := flow.MarshalText()
+		if err != nil {
+			return nil, err
+		}
+		args = append(args, string(fb))
+	}
+	return args, nil
+}
+
+// DumpFlowsWithFlowArgs retrieves statistics about all flows for the specified bridge.
+// If a table has no active flows and has not been used for a lookup or matched
+// by an incoming packet, it is filtered from the output.
+// We neeed to add a Matchflow to filter the dumpflow results. For example filter based on table, cookie.
+func (o *OpenFlowService) DumpFlowsWithFlowArgs(bridge string, flow *MatchFlow) ([]*Flow, error) {
+	args := []string{"dump-flows", bridge}
+	args, err := addArgsForDumpFlows(args, o.c.ofctlFlags, flow)
+	if err != nil {
+		return nil, err
+	}
+	return o.dumpFlows(args)
```

Comment:
with the below, this would be simplified to just:
```golang
return o.dumpFlows(bridge, flow)
```

Reply:
Do you mean this?
```go
return o.DumpFlowsWithFlowArgs(bridge, flow)
```

Reply:
yes; i realized halfway through writing these comments that `DumpFlows` could just be a wrapper around `DumpFlowsWithFlowArgs` instead of an unexporter `dumpFlows` helper, and i forgot to update my comments accordingly 😅

Reply:
Done